### ISSUE TITLE
Implement simple path measurements and path selection

### ISSUE THREAD

**Issue by mlegner** [MystenLabs/scion-rs#55]
- [ ] Measure aliveness and latency of paths
  - In the Go implementation, this is done via a traceroute to the final hop, see [here](https://github.com/scionproto/scion/blob/9f77b13ed8b36273de21f72d01d67bb0cd301e4e/private/app/path/pathprobe/paths.go#L139-L284)
- [ ] Choose path based on latency, path length (?)
- [ ] Ideally shared among multiple sockets/connections of the same application
- [ ] Potential optimization: Use RTT measurements of upper-layer protocol (i.e., QUIC).
- [ ] Question: Should this instead be handled by the SCION Daemon?
- [ ] Question: How to choose between IP and SCION?
      - Always choose SCION if it is available?
      - Compare performance?

Probably depends on #54

**Comment by jpcsmith** [comment]
@mlegner The pan implementation in golang seems to use [Ping instead of traceroute](https://github.com/netsec-ethz/scion-apps/blob/3afc9a9118080aa78e6a6435c06549a8a6c0bd23/_examples/helloquic/helloquic.go#L107C6-L107C15) to select paths. Do we want to use the same approach?

**Comment by mlegner** [comment]
> The pan implementation in golang seems to use [Ping instead of traceroute](https://github.com/netsec-ethz/scion-apps/blob/3afc9a9118080aa78e6a6435c06549a8a6c0bd23/_examples/helloquic/helloquic.go#L107C6-L107C15) to select paths. Do we want to use the same approach?

Yes, I became aware of that as well. The aliveness check (e.g., for `scion showpaths`) uses traceroute while the path selection in pan uses pings. In the current deployments, there is probably very little difference between the two in terms of measurement results as we usually have very small source and destination ASes.

I don't have a strong opinion but would probably go with the traceroute approach to start with.
